src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -370,6 +376,19 @@ private static void verifyConfig(Configuration testConfig, Configuration parsedC
                     .that(inputConfiguration.getDefaultPropertyValue(property))
                     .isEqualTo(testConfig.getAttribute(property));
         }
+        final Set<Integer> expectedViolationLines = new TreeSet<>();
+        for (String violationMessage : expected) {
+            expectedViolationLines.add(
+                    Integer.parseInt(
+                            violationMessage
+                                    .substring(0, violationMessage.indexOf(':'))
```

Comment:
please explain this code.
What is role of ":" ?

Reply:
we want to extract the line number out of a violation message which follows this structure
``` "4:12: check message goes here"```
The ":" is the one which follows the line number.